spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -606,8 +582,12 @@ contract PerpetualTranche is
         external
         override
         afterStateUpdate
-        returns (IERC20Upgradeable[] memory, uint256[] memory)
+        returns (TokenAmount[] memory)
     {
+        uint256 perpSupply = totalSupply();
+        if(perpSupply == 0){
+            revert UnacceptableRedemption();
+        }
```

Comment:
Change the below line to use `perpSupply`, instead of re-calling `totalSupply()`?